api.lua

```diff
@@ -3,17 +3,48 @@
 function areas:getAreasAtPos(pos)
 	local a = {}
 	local px, py, pz = pos.x, pos.y, pos.z
-	for id, area in pairs(self.areas) do
-		local ap1, ap2 = area.pos1, area.pos2
-		if px >= ap1.x and px <= ap2.x and
-		   py >= ap1.y and py <= ap2.y and
-		   pz >= ap1.z and pz <= ap2.z then
-			a[id] = area
+	if self.store then
+		local areas = self.store:get_areas_for_pos(pos, false, true)
+		for store_id, store_area in pairs(areas) do
+			local id = tonumber(store_area.data)
+			a[id] = self.areas[id]
+		end
+	else
+		for id, area in pairs(self.areas) do
```

Comment:
This does seem to be faster, but only by about 1-8ms per million entries.  I don't think wee need to optimize it that much.